pkg/controller/performanceprofile/components/tuned/tuned.go

```diff
@@ -68,23 +90,12 @@ func NewNodePerformance(assetsDir string, profile *performancev1alpha1.Performan
 		},
 	}
 
-	// we should sort our matches, otherwise we can not predict the order of nested matches
-	sortedKeys := []string{}
-	for k := range profile.Spec.NodeSelector {
-		sortedKeys = append(sortedKeys, k)
-	}
-	sort.Strings(sortedKeys)
-
 	priority := uint64(30)
-	copyNodeSelector := map[string]string{}
-	for k, v := range profile.Spec.NodeSelector {
-		copyNodeSelector[k] = v
-	}
 	recommends := []tunedv1.TunedRecommend{
 		{
-			Profile:  &name,
-			Priority: &priority,
-			Match:    getProfileMatches(sortedKeys, copyNodeSelector),
+			Profile:             &name,
+			Priority:            &priority,
+			MachineConfigLabels: componentsprofile.GetMachineConfigLabel(profile),
```

Comment:
Did they remove the nested matches?

Reply:
no, it still remains (so there is no concern for backwards compatibility break) but this is how we will use it for our need from now on.